### ISSUE TITLE
 How to train semantic segmentation head with detectron2?

### ISSUE THREAD

**Issue by animareversed** [facebookresearch/detectron2#500]
Hi, 
I really don't understand what kind of images should I pass to segmentation head. I checked coco images (stuff annotations images) and they are in 'P' mode while cityscapes images are 'RGB'.
So,
should I feed the head with colored images or the images with label ID (gratscale)?
I tried to feed it with colored images but with just one channel because the Cross entropy function does not accept multi channel arrays.

Thanks in advance

**Comment by ppwwyyxx** [comment]
https://detectron2.readthedocs.io/tutorials/models.html#model-input-format

labelId needs to be integers so the segmentation is one-channel.

**Comment by animareversed** [comment]
Ok, thank you... But, obviously labelids are integers the question was images have to be colored or labelled?
For example, let's take two cityscapes images:

labelled:
![aachen_000000_000019_gtFine_color](https://user-images.githubusercontent.com/55583897/70607290-719d5180-1bfe-11ea-9b7d-a9d836abc166.png)


colored:
![aachen_000000_000019_gtFine_labelIds](https://user-images.githubusercontent.com/55583897/70607268-64806280-1bfe-11ea-890c-123939614c1c.png)

both are in RGB so they have 3 channels.
Thanks

**Comment by ppwwyyxx** [comment]
The ground truth of the semantic segmentation head is a one-channel image with integer values.

**Comment by ugamman** [comment]
Hi, 
I am facing the same problem. In my case I am trying to use COCO dataset.
I read and printed the values of a segmentation image ground truth and I obtained these:
[0,  92,  96, 132, 140, 142, 149, 164, 167, 172, 181, 183]
but you said that they should be in that range. So. what do I have to do?
Thank you in advance.

**Comment by ppwwyyxx** [comment]
You can convert them to the range.

It means class "255" will be ignored and won't be trained.

**Comment by mukulkhanna** [comment]
@animareversed Which model did you use for training this semantic head? Thank you
